scripts/teams-bulk-create-teams/README.md

```diff
@@ -79,6 +79,67 @@ end{
 
 [!INCLUDE [More about PnP PowerShell](../../docfx/includes/MORE-PNPPS.md)]
 
+# [CLI for Microsoft 365](#tab/cli-m365-ps)
+```powershell
+# Usage example:
+# .\Create-BulkTeams.ps1 -OwnerEmail "user@contoso.com"
+
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $true)]
+    [string]$OwnerEmail,
+    [Parameter(Mandatory = $false)]
+    [string] $SiteListJsonFile = ".\teams.json"
+)
+begin {
+    #Log in to Microsoft 365
+    Write-Host "Connecting to Tenant" -f Yellow
+
+    $m365Status = m365 status
+    if ($m365Status -match "Logged Out") {
+        m365 login
+    }
+
+    Write-Host "Connection Successful!" -f Green
+
+    $sites = Get-Content $SiteListJsonFile -Raw | ConvertFrom-Json
+    $prefixes = "HR", "Finance", "ICT", "Service Desk", "Client Services", "Project Alpha", "Project Beta", "Project Charlie", "Leadership", "Community"
+}
+process {
+    $prefixes | Foreach-Object {
+        $prefix = $_
+
+        $sites | Foreach-Object {
+            $siteTitle = "$($_.SiteTitle.Replace("XXXXX", $prefix))"
+            $mailNickname = "$($_.SiteUrl.Replace("XXXXX", $prefix).Replace(" ",''))"
+
+            # Check if team exists
+            $team = m365 teams team get --name $siteTitle | ConvertFrom-Json
+
+            if (-not $team) {
+                # Create a Microsoft 365 Group
+                Write-Host "Creating group $($siteTitle)"
+                m365 aad o365group add --displayName "$($siteTitle)" --description "Testing Site for $($siteTitle)" --mailNickname $mailNickname --owners "$($OwnerEmail)" --isPrivate (!$_.IsPublic).ToString().ToLowerInvariant()
+
+                # Wait for group creation to complete
+                Start-sleep -Seconds 10
+
+                # Create a new Microsoft Teams team under existing Microsoft 365 group
```

Comment:
the problem here is that we expect that the aad group will provision internally in 10 s. That may not be the case and then the next command `aad o365group teamify` will fail with error 
```
Error: Failed to execute MS Graph backend request GetGroupInternalApiRequest. The server failed to respond correctly. Response Code: NotFound, Reason: Reason: Not Found, Correlation headers: request-id: 50882430-dd6a-4c05-a39c-8f81e4f35e84..
```
we could solve this by (both options are ok for me):
- lets add a more extensive comment here describing that this kind of issue might happen and in this case the sleep may be extended to overcome this
- we should add a helper function that checks if the group is already provisioned in a while loop. The check could be done every 10 s and with a max restriction to not check more than 100 times. If after 100 x 10 s the group is still not provisioned we may stop the script with fail.

Reply:
Thanks for the review comments @Adam-it 
I will prefer to go with option 2 - add a helper function that checks if the group is already provisioned in a while loop